jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1822,31 +1846,40 @@ YUI.add('juju-view-utils', function(Y) {
   */
   utils.generateServiceName = function(
     charmName, charmId, services, ignoreSelf) {
-    var charmCount = 0;
     // Loop through each service and grab its charmid to see if it matches.
-    services.each((service) => {
+    var highestCounter = Math.max.apply(Math, services.map((service) => {
       // Remove the version number from the charm id as we don't care about the
       // version when comparing charms.
       var serviceCharmId = utils._extractCharmName(service.get('charm'));
       var providedCharmId = utils._extractCharmName(charmId);
       // If we have a matching charm then increase the count.
       if (serviceCharmId === providedCharmId) {
-        charmCount += 1;
+        // Remove the service name and dash from the name to get the counter.
+        var counter = service.get('name').replace(
```

Comment:
why not split on last - and grab the last block? I figure this would be less error prone.